src/rules.jl

```diff
@@ -60,6 +60,7 @@
 @define_diffrule Base.lgamma(x)               = :(  digamma($x)                        )
 @define_diffrule Base.Math.JuliaLibm.log1p(x) = :(  inv($x + 1)                        )
 @define_diffrule Base.transpose(x)            = :(  1                                  )
+@define_diffrule Base.abs(x)                  = :( $x > 0 ? one($x) : -one($x)         )
```

Comment:
Should we use `signbit` instead here (since, e.g. `!(-0.0 < 0.0)`)? At least, this is what ForwardDiff's manual definition currently uses, though IIRC that decision was made fairly arbitrarily at the time...

Reply:
That makes sense. Will make the change.